alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php

```diff
@@ -865,4 +901,54 @@ public function getRequestParameter($name, $default = null)
 		$returnValue = parent::getRequestParameter($name, $default);
 		return $returnValue ? $returnValue : $default;
 	}
+	
+	protected function saveCurrentBundleCacheKey($context)
+	{
+		//Avoid updating current cache hash on the ui conf if dynamic params affecting the result are being sent
+		$versions = $this->getRequestParameter(self::VERSIONS_PARAM_NAME);
+		if($versions)
+		{
+			return;
+		}
+		
+		if($context->bundle_name != $context->uiConf->getCurrentCacheKey())
+		{
+			//Avoid system load by controlling the percentage of ui confs how will get this update
+			if(rand(0, 100) < kConf::get("current_version_save_ration", kConfMapNames::EMBED_PLAYKIT, 30))
+			{
+				$context->uiConf->setCurrentCacheKey($context->bundle_name);
+				$context->uiConf->save();
+			}
+			else
+			{
+				KalturaLog::debug("Skipping ui conf update to avoid unnecessary system load");
+			}
+		}
+	}
+	
+	protected function tryServingExistingCacheVersion($context, $errCode, $message = null)
+	{
+		$existingCacheVersion = $context->uiConf->getCurrentCacheKey();
```

Comment:
general note - 
Saving current cache key on UI conf is not essential, as the current cache key can be calculated 'on the spot' from the versions that are currently required by it ( hash( ui confi versions ) while the action of saving new version on UICONF in DB, will create load on DB any time new version will be released.

Reply:
The entire point is that the current hash of versions is not in the cache, this is why we got to this point in the first place. I need the old hash of the UI conf versions, which I currently don't have a way to get. If in the future we will have the old version pam also stored we can re-visit this.